internal/impl/amqp09/integration_test.go

```diff
@@ -64,27 +79,82 @@ input:
     queue: queue-$ID
     queue_declare:
       durable: true
-      enabled: true
+      enabled: true%s
     bindings_declare:
       - exchange: exchange-$ID
         key: benthos-key
-`
+`,
+			queueDeclareAutoDeleteFragment,
+		)
+	}
+
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	pool.MaxWait = time.Second * 30
+
+	resource, err := pool.Run("rabbitmq", "management", nil)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, pool.Purge(resource))
+	})
 
-	suite := integration.StreamTests(
+	_ = resource.Expire(900)
+	require.NoError(t, pool.Retry(func() error {
+		client, err := amqp.Dial(fmt.Sprintf("amqp://guest:guest@localhost:%v/", resource.GetPort("5672/tcp")))
+		if err == nil {
+			_ = client.Close()
+		}
+		return err
+	}))
+
+	commonTests := integration.StreamTests(
 		integration.StreamTestOpenClose(),
 		integration.StreamTestMetadata(),
 		integration.StreamTestMetadataFilter(),
 		integration.StreamTestSendBatch(10),
 		integration.StreamTestStreamSequential(1000),
 		integration.StreamTestStreamParallel(1000),
+	)
+
+	lossyTests := integration.StreamTests(
```

Comment:
Hmm- on thinking about it (and testing in my real environment), the fact that the lossy tests fail in this configuration is actually problematic.

Lossy connections are a fact of life and we want Benthos to keep consuming as per config in that event.

I'll change the tests so that all suites run for both config variants and look what what I can do with the AMQP lifecycle to ensure we re-establish the queues and bindings on reconnect when auto-delete is set.